envoy/network/transport_socket.h

```diff
@@ -263,6 +263,11 @@ class TransportSocketFactory {
    */
   virtual bool supportsAlpn() const { return false; }
 
+  /**
+   * Returns the default SNI for transport sockets created by this factory.
```

Comment:
In my "I come from Chrome" world, I think of a SocketFactory as a thing which returns "Socket" object on which one calls Connect() and passes in the remote host. So the idea of the *Factory* having a default SNI continues to blow my mind :)

But it's clearly a thing in Envoy, of course, so this makes sense. That being said, it might be nice to have a comment here which explains when/if the returned value can be empty. It would probably be a good idea to mention that this is the SNI being sent, so this will only be present on client sockets, I think?

Reply:
done